controllers/networkplugins_controller.go

```diff
@@ -17,35 +17,480 @@ limitations under the License.
+	currentSpec *plumberv1.NetworkPluginsSpec
+	prevSpec    *plumberv1.NetworkPluginsSpec
+}
+
+type MultusT plumberv1.Multus
+type WhereaboutsT plumberv1.Whereabouts
+type SriovT plumberv1.Sriov
+
+type ApplyPlugin interface {
+	ParseTemplate(string) error
+	ApplyTemplate(string) error
+}
+
+func createFile(config map[string]interface{}, t *template.Template, filename string) error {
+	f, err := os.Create(filename)
+	if err != nil {
+		return err
+	}
+	err = t.Execute(f, config)
+	if err != nil {
+		fmt.Printf("template.Execute failed: %s\n", err)
+		f.Close()
+		return err
+	}
+	f.Close()
+	return nil
+}
+
+func (multusConfig *MultusT) ParseTemplate(outputDir string) error {
+	config := make(map[string]interface{})
+	if multusConfig.Namespace != "" {
+		config["Namespace"] = multusConfig.Namespace
+	} else {
+		config["Namespace"] = DEFAULT_NAMESPACE
+	}
+
+	if multusConfig.MultusImage != "" {
+		config["MultusImage"] = multusConfig.MultusImage
+	} else {
+		config["MultusImage"] = MULTUS_IMAGE
+	}
+
+	t, err := template.ParseFiles(TEMPLATE_DIR + "multus/multus.yaml")
+	if err != nil {
+		fmt.Printf("ERROR PARSEFILE: %v\n", err)
+		return err
+	}
+
+	if err := createFile(config, t, outputDir+"multus.yaml"); err != nil {
+		return err
+	}
+	return nil
+}
+
+func (multusConfig *MultusT) ApplyTemplate(outputDir string) error {
+	fmt.Println("Applying Multus")
+	return nil
+}
+
+func (whereaboutsConfig *WhereaboutsT) ParseTemplate(outputDir string) error {
+	config := make(map[string]interface{})
+	fmt.Println("Rendering Whereabouts...")
+	if whereaboutsConfig.Namespace != "" {
+		config["Namespace"] = whereaboutsConfig.Namespace
+	} else {
+		config["Namespace"] = DEFAULT_NAMESPACE
+	}
+
+	if whereaboutsConfig.WhereaboutsImage != "" {
+		config["WhereaboutsImage"] = whereaboutsConfig.WhereaboutsImage
+	} else {
+		config["WhereaboutsImage"] = WHEREABOUTS_IMAGE
+	}
+
+	t, err := template.ParseFiles(TEMPLATE_DIR + "whereabouts/whereabouts.yaml")
+	if err != nil {
+		fmt.Printf("ERROR PARSEFILE: %v\n", err)
+		return err
+	}
+
+	if err := createFile(config, t, outputDir+"whereabouts.yaml"); err != nil {
+		return err
+	}
+	return nil
+}
+
+func (whereaboutsConfig *WhereaboutsT) ApplyTemplate(outputDir string) error {
+	fmt.Println("Applying Whereabouts")
+	return nil
+}
+
+func (sriovConfig *SriovT) ParseTemplate(outputDir string) error {
+	config := make(map[string]interface{})
+	fmt.Println("Rendering Sriov...")
+
+	if sriovConfig.Namespace != "" {
+		config["Namespace"] = sriovConfig.Namespace
+	} else {
+		config["Namespace"] = DEFAULT_NAMESPACE
+	}
+
+	if sriovConfig.SriovCniImage != "" {
+		config["SriovCniImage"] = sriovConfig.SriovCniImage
+	} else {
+		config["SriovCniImage"] = SRIOV_CNI_IMAGE
+	}
+
+	if sriovConfig.SriovDpImage != "" {
+		config["SriovDpImage"] = sriovConfig.SriovDpImage
+	} else {
+		config["SriovDpImage"] = SRIOV_DP_IMAGE
+	}
+
+	// Apply the SRIOV CNI
+	t, err := template.ParseFiles(TEMPLATE_DIR + "sriov/sriov-cni.yaml")
```

Comment:
The common/safe way to work with paths:
```suggestion
	t, err := template.ParseFiles(path.Join(TEMPLATE_DIR, "sriov/sriov-cni.yaml"))
```
Holds for all cases were you are building paths.